### PR TITLE
[SPARK-40466][SS] Improve the error message when DSv2 is disabled while DSv1 is not avaliable

### PR DESCRIPTION
**Pull request by huanliwang-db**
…le DSv1 is not avaliable.

### What changes were proposed in this pull request?

Improve the error message when DSv2 is disable while its fallback DSv1 is not available.

### Why are the changes needed?

Improve the user experience. When users get UnsupportOperationError for the disabled DSv2, they are able to know which config to modify to enable the V2 source.

### Does this PR introduce _any_ user-facing change?

Yes, error message.

### How was this patch tested?

N/A, just the message change

### CODE REVIEW

**Comment by huanliwang-db** [comment]
@HeartSaVioR Please review this change

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 1]
```diff
@@ -1599,9 +1599,18 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
...
+  def microBatchUnsupportedByDataSourceError(
+      srcName: String,
+      disabledSources: String,
+      table: Table): Throwable = {
     new UnsupportedOperationException(
-      s"Data source $srcName does not support microbatch processing.")
+      s"""
+         |Data source $srcName does not support microbatch processing.
+         |
+         |It could be the data source is disabled at spark.sql.streaming.disabledV2MicroBatchReaders
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:1610]: Should better use `SQLConf.get.DISABLED_V2_STREAMING_MICROBATCH_READERS.key`

Comment (HeartSaVioR) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:1610]: It could be -> Either

Comment (huanliwang-db) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala]: Done

Comment (huanliwang-db) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala]: Done

**Review by HeartSaVioR — APPROVED** [review]
+1

I thought of more direct message (different error message per case), but given that the config is not something majorly used from end users, this seems to be good enough.

Thanks for the work!

**Comment by HeartSaVioR** [comment]
Thanks! Merging to master.

**Pull request closed without merging**